pnpm/test/monorepo/dedupePeers.test.ts

```diff
@@ -128,19 +129,22 @@ auto-install-peers=false`, 'utf8')
   await addDistTag({ package: '@pnpm.e2e/foo', version: '2.0.0', distTag: 'latest' })
   await addDistTag({ package: '@pnpm.e2e/bar', version: '100.1.0', distTag: 'latest' })
 
-  await execPnpm(['update', '--filter', 'project-2', '--latest'])
+  // update foo only for project-2
+  await execPnpm(['update', '--filter', 'project-2', '--latest', '@pnpm.e2e/foo'])
 
-  // project 1's manifest is unaffected, while project 2 has foo updated
+  // project 1's manifest is unaffected, while project 2 has only foo updated
   expect(loadJsonFile<any>('project-1/package.json').dependencies['@pnpm.e2e/foo']).toBe('1.0.0') // eslint-disable-line
   expect(loadJsonFile<any>('project-1/package.json').dependencies['@pnpm.e2e/bar']).toBe('100.0.0') // eslint-disable-line
   expect(loadJsonFile<any>('project-2/package.json').dependencies['@pnpm.e2e/foo']).toBe('2.0.0') // eslint-disable-line
+  expect(loadJsonFile<any>('project-2/package.json').dependencies['@pnpm.e2e/bar']).toBe('100.0.0') // eslint-disable-line
```

Comment:
I didn't notice this issue earlier, or at work (changing unrelated workspace projects was the first thing we bumped into, so our reproductions focused on that; a bit of tunnel-vision on my behalf 😓), but it came up while investigating/reproducing https://github.com/pnpm/pnpm/issues/8661